addons/point_of_sale/models/account_journal.py

```diff
@@ -15,6 +15,25 @@ def _check_type(self):
         if methods:
             raise ValidationError(_("This journal is associated with a payment method. You cannot modify its type"))
 
+    def has_active_payments(self):
```

Comment:
```suggestion
    def _check_no_active_payments(self):
```
I would suggest factorizing everything into a method that raises the ValidationError, as it's unlikely we'll need to use this method outside of performing the check on archiving/deleting a journal

Also, let's make the method private since it doesn't need to be called by jsonrpc.